src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action['type']) {
```

Comment:
I guess, it's better to use dot notation here, because `type` is not a variable. ( `action.type`).